### ISSUE TITLE
E2523 Only one of [VirtualName, Ebs, NoDevice] should be specified for Resources/LaunchTemplate/Properties/LaunchTemplateData/BlockDeviceMappings/0

### ISSUE THREAD

**Issue by ARMAAN7139** [aws-cloudformation/cfn-python-lint#1614]
cfn-lint 0.33.0

```
Resources:
  LaunchTemplate:
    Type: AWS::EC2::LaunchTemplate
    Properties:
      LaunchTemplateData:
        BlockDeviceMappings:
          - DeviceName: DeviceName
          - Ebs:
              DeleteOnTermination: false
```

**Comment by PatMyron** [comment]
Reproduced, thanks for the template! Will need to look into this type and rule:

[`AWS::EC2::LaunchTemplate.BlockDeviceMapping`](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-ec2-launchtemplate-blockdevicemapping.html)

https://github.com/aws-cloudformation/cfn-python-lint/blob/fe0af928a4ab26a6cedfd76564a26c9a4120b14f/src/cfnlint/rules/resources/properties/OnlyOne.py#L13

If you need an immediate workaround, [individual rules can be ignored at the resource-level:](https://github.com/aws-cloudformation/cfn-python-lint#resource-based-metadata)
```yaml
    Metadata:
      cfn-lint:
        config:
          ignore_checks:
          - E2523
```

**Comment by kddejong** [comment]
We may need to where that is defined.  Since there are two block device mappings its complaining about the first one.
Is only `DeviceName: DeviceName` acceptable?  What do you get when that is deployed?  I thought if you didn't actually want a device you still needed `NoDevice` set.

**Comment by PatMyron** [comment]
think the problem's this line, will send a PR out later:
https://github.com/aws-cloudformation/cfn-python-lint/blob/fe0af928a4ab26a6cedfd76564a26c9a4120b14f/src/cfnlint/rules/resources/properties/OnlyOne.py#L45

**Comment by kddejong** [comment]
For the first BlockDeviceMapping
```
DeviceName: DeviceName
```
This still works and completes.  You are able to override the value which you will need to do to make it work.  Additionally as is (without overrides) it will error that ebs isn't specified.  When using the GUI and adding a size it seemed to be fine. 

Also when trying to use this launch template this BlockDeviceMapping 
```
- Ebs:
     DeleteOnTermination: false
```
It errors with no DeviceName which I couldn't overcome.  @ARMAAN7139 beyond this template succeeding are you able to launch instances from it?  If so how are you doing it?  Are you just overriding the values?

I think the override capabilities are why we should remove these constraints for launch templates.  It may leave the LaunchTemplate in a weird spot but its never really "broken" because you can override.

**Comment by ARMAAN7139** [comment]
Hello,  I am able to provision the stack either  properties since they are not require  properties for the resource. But I can not specify the  `DeviceName` and  Either one of the following `Ebs`, `NoDevice`  and  String `VirtualName`.  If you look at the example for resource  launch configuration `BlockDeviceMapping`  properties, it is same property and CfnLint does not complain about that block of code. Thank you  and apologies for my English.
